lightning/src/ln/onchaintx.rs

```diff
@@ -657,12 +663,20 @@ impl<ChanSigner: ChannelKeys> OnchainTxHandler<ChanSigner> {
 		None
 	}
 
-	pub(crate) fn block_connected<B: Deref, F: Deref, L: Deref>(&mut self, txn_matched: &[&Transaction], claimable_outpoints: Vec<ClaimRequest>, height: u32, broadcaster: B, fee_estimator: F, logger: L)
+	/// Upon channelmonitor.block_connected(..) or upon provision of a preimage on the forward link
+	/// for this channel, provide new relevant on-chain transactions and/or new claim requests.
+	/// Formerly this was named `block_connected`, but it is now also used for claiming an HTLC output
+	/// if we receive a preimage after force-close.
+	pub(crate) fn update_claims_view<B: Deref, F: Deref, L: Deref>(&mut self, txn_matched: &[&Transaction], claimable_outpoints: Vec<ClaimRequest>, latest_height: Option<u32>, broadcaster: &B, fee_estimator: &F, logger: &L)
 		where B::Target: BroadcasterInterface,
 		      F::Target: FeeEstimator,
 					L::Target: Logger,
 	{
-		log_trace!(logger, "Block at height {} connected with {} claim requests", height, claimable_outpoints.len());
+		let height = match latest_height {
+			Some(h) => h,
+			None => self.latest_height,
+		};
```

Comment:
Good solution to fulfill height with latest block seen. I was surprised at first it didn't break tests but now that I remember all the txn rebroadcast are moving forward only in case of block connections/disconnections.

Reply:
great, thank you!